apps/prairielearn/assets/scripts/question.ts

```diff
@@ -113,58 +115,22 @@ function handleStatusChange(socket: Socket, msg: StatusMessage) {
 }
 
 function fetchResults(socket: Socket, submissionId: string) {
-  const questionContainer = document.querySelector<HTMLElement>('.question-container');
-
-  if (!questionContainer) return;
-
-  const {
-    variantId,
-    questionId,
-    instanceQuestionId,
-    userId,
-    variantToken,
-    urlPrefix,
-    questionContext,
-    csrfToken,
-    authorizedEdit,
-  } = questionContainer.dataset;
-
   const modal = $('#submissionInfoModal-' + submissionId);
   const wasModalOpen = (modal.data('bs.modal') || {})._isShown;
   modal.modal('hide');
 
-  socket.emit(
-    'getResults',
-    {
-      question_id: questionId,
-      instance_question_id: instanceQuestionId === '' ? null : instanceQuestionId,
-      variant_id: variantId,
-      user_id: userId,
-      variant_token: variantToken,
-      submission_id: submissionId,
-      url_prefix: urlPrefix,
-      question_context: questionContext,
-      csrf_token: csrfToken,
-      // Indicates whether submissions are allowed, either because
-      // the instance question is part of the current user's
-      // assessment instance (authorized_edit==true) or because the
-      // question is open in preview mode (authz_result==undefined)
-      authorized_edit: authorizedEdit === 'true',
-    },
-    function (msg: SubmissionPanels | null) {
-      // We're done with the socket for this incarnation of the page
-      socket.close();
-      if (msg) {
-        updateDynamicPanels(msg, submissionId);
-      } else {
-        console.error(`Error retrieving results for submission ${submissionId}`);
-      }
-      // Restore modal state if need be
-      if (wasModalOpen) {
-        $('#submissionInfoModal-' + submissionId).modal('show');
-      }
-    },
-  );
+  const submissionPanel = document.getElementById('submission-' + submissionId);
+  if (!submissionPanel) return;
+
+  const submissionBody = submissionPanel.querySelector<HTMLDivElement>('.js-submission-body');
+  if (!submissionBody) return;
+
+  loadPendingSubmissionPanel.call(submissionBody, true).finally(() => {
+    // Restore modal state if need be
+    if (wasModalOpen) {
+      $('#submissionInfoModal-' + submissionId).modal('show');
+    }
```

Comment:
I wonder if I can push this down into `loadPendingSubmissionPanel`. That way, `loadPendingSubmissionPanel` doesn't even need to return a Promise, which would simplify calling that function.

Reply:
From my experience, restoring the modal does't actually work right now. If I open the modal before grading is complete, it will be closed (but not reopened) when the grading completes. So I wouldn't fret too much about this functionality, unless you want to fix this.

Reply:
I ended up ripping out the modal restoration stuff entirely; we don't need to be quite as clever as I clearly thought we did 7 years ago or whatever 😂